#Dictionary

In [5]:
# 1. Defining a student using a List (positional data)
student_list = ["Rahul", 22, 91.5, "Nagpur"]

In [6]:

# 2. Defining a student using a Dictionary (key-value mapping)
student = {
    "name": "Rahul",
    "age": 22,
    "CGPA": 91.5,
    "city": "Nagpur"
}

In [7]:
# 3. Printing the dictionary
print(student)

{'name': 'Rahul', 'age': 22, 'CGPA': 91.5, 'city': 'Nagpur'}


In [8]:
# 4. Checking the data type
print(type(student))

<class 'dict'>


In [21]:
# student[1]

In [10]:
student["name"]

'Rahul'

In [11]:
student["state"] = "Maharastra"

In [12]:
student

{'name': 'Rahul',
 'age': 22,
 'CGPA': 91.5,
 'city': 'Nagpur',
 'state': 'Maharastra'}

In [13]:
student.get("adress", "Key not found!")

'Key not found!'

In [14]:
student.get("age", "Key not found")

22

In [15]:
student.pop("state")

'Maharastra'

In [16]:
student

{'name': 'Rahul', 'age': 22, 'CGPA': 91.5, 'city': 'Nagpur'}

In [17]:
student

{'name': 'Rahul', 'age': 22, 'CGPA': 91.5, 'city': 'Nagpur'}

In [24]:
marks = {
    "Math": 90,
    "Science": 80,
    "English": 70
}

total = 0

for subject in marks.values():
    total += subject

print(total / len(marks))

80.0


In [27]:
employee = {
    "salary": 50000
}
for k in employee.values():
  k = k+(k*0.1)
print(k)

55000.0


In [28]:
employee = {
    "salary": 50000
}
employee["salary"] = employee["salary"] *1.1
print(employee["salary"])

55000.00000000001


### Note on Floating Point Precision
Computers store numbers in binary (base-2). Just as the fraction $1/3$ cannot be represented exactly in decimal ($0.333...$), the decimal $0.1$ or $1.1$ cannot be represented exactly in binary. This leads to tiny rounding differences like `55000.00000000001` instead of `55000.0`.

To avoid this in real applications (like money), you can use the `round()` function:
```python
print(round(employee['salary'], 2))
```